pkg/serverstate/statetest/test_runner_ondemand.go

```diff
@@ -142,3 +143,37 @@ func TestOnDemandRunnerConfig(t *testing.T, factory Factory, restartF RestartFac
 		}
 	})
 }
+
+func TestOnDemandRunnerConfig_LabelTargeting(t *testing.T, factory Factory, restartF RestartFactory) {
+	require := require.New(t)
+
+	s := factory(t)
+	defer s.Close()
+
+	labels := map[string]string{"name": "testrunner", "env": "test"}
+
+	// Set
+	rec := serverptypes.TestOnDemandRunnerConfig(t, &pb.OnDemandRunnerConfig{
+		Name:   "test",
+		OciUrl: "h/w:s",
+		TargetRunner: &pb.Ref_Runner{
+			Target: &pb.Ref_Runner_Labels{
+				Labels: &pb.Ref_RunnerLabels{
+					Labels: labels,
+				},
+			},
+		},
+	})
+
+	err := s.OnDemandRunnerConfigPut(rec)
+	require.NoError(err)
+
+	resp, err := s.OnDemandRunnerConfigGet(&pb.Ref_OnDemandRunnerConfig{
+		Id: rec.Id,
+	})
+	require.NoError(err)
+	require.NotNil(resp)
+
+	// Ensure the target saved properly
+	require.Equal(rec.TargetRunner.Target, resp.TargetRunner.Target)
```

Comment:
I think this go-test failure is happening here? We might have to compare the actual map string string, since I don't think go likes comparing two protos of `pb.Ref_Runner_Labels { ... }`

Reply:
Yeah, be very careful when comparing proto structures, cause they often contain other state. In general, I avoid it.